src/components/EventForm/EventForm.tsx

```diff
@@ -0,0 +1,271 @@
+import { ChangeEvent } from "react";
+import Button from "../Button/Button";
+import InputBox from "../InputBox/InputBox";
+import { addDoc, collection, Timestamp } from "firebase/firestore";
+import { db } from "../../firebase";
+import { useState } from "react";
+import "./EventForm.scss";
+import { useNavigate } from "react-router-dom";
+import { isBefore } from "date-fns";
+
+type DefaultForm = {
+  eventName: string;
+  eventCategory: string;
+  eventCapacity: number | null;
+  eventDescription: string;
+  eventImages: string[];
+  eventDate: Timestamp | null;
+};
+
+type EventFormProps = {
+  handleNewEvent: () => void;
+};
+
+const EventForm = ({ handleNewEvent }: EventFormProps) => {
+  const [databaseInput, setDatabaseInput] = useState<DefaultForm>({
+    eventName: "",
+    eventCategory: "",
+    eventCapacity: null,
+    eventDescription: "",
+    eventDate: null,
+    eventImages: [""],
+  });
+  const [images, setImages] = useState<string[]>([]);
+  const [inputValue, setInputValue] = useState<string>("");
+  const [required, setRequired] = useState({
```

Comment:
I think this can be named better, something where the keys describe the error. 

```tsx
  const [ validationError, setValidationError] = useState({
    eventName: true,
    eventCategory: true,
    eventDate: true,
    // ETC
  });
```